app/menu.js

```diff
@@ -2,16 +2,41 @@
  * Import/require this file to set up the Menu of the windows
  */
 
-import electron, { BrowserWindow, Menu, app } from 'electron'
+import electron, { BrowserWindow, Menu, app, dialog } from 'electron'
 
 import ImportWindow from './windows/Import/window'
 import SettingsWindow from './windows/Settings/window'
 import ResolveIPNS from './windows/ResolveIPNS/window'
 import PublishToIPNS from './windows/PublishToIPNS/window'
 
+import { addFilesPaths } from './windows/Storage/fileIntegration'
+
 const template = [{
   label: 'File',
   submenu: [{
+    label: 'Add File',
+    click () {
+      const selectOptions = {
+        title: 'Add File',
+        properties: ['openFile', 'multiSelections']
+      }
+
+      const paths = dialog.showOpenDialog(app.mainWindow, selectOptions)
+      addFilesPaths(paths)
+    }
+  }, {
+    label: 'Add Directory',
+    click () {
+      const selectOptions = {
+        title: 'Add Directory',
+        properties: ['openDirectory', 'multiSelections']
+      }
+
+      const paths = dialog.showOpenDialog(selectOptions)
```

Comment:
Why this doesn't have app.mainWindow and the other has?

Reply:
This is ran before the StorageList is created. Maybe is worth considering moving this to StorageList and creating the menu afterwards? We disable the menu for any other window anyway...

Reply:
I see, as we discussed maybe move the creation of the Menu into the Storage Window only.

Reply:
Turns out we can only create the menu from the main process. Nonetheless, I added the missing `app.mainWindow` to the dialog and it seems to work (StorageList is unusable while the dialog is open).

The warning I was getting: "Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged." seems to be unrelated to this.